drivers/timer/arcv2_timer0.c

```diff
@@ -190,13 +190,22 @@ static u32_t elapsed(void)
 	} while (timer0_count_register_get() < val);
 
 	if (ctrl & _ARC_V2_TMR_CTRL_IP) {
-		overflow_cyc += last_load;
+		overflow_cycles += last_load;
 		/* clear the IP bit of the control register */
 		timer0_control_register_set(_ARC_V2_TMR_CTRL_NH |
 					    _ARC_V2_TMR_CTRL_IE);
+		/* use sw triggered irq to remember the timer irq request
+		 * which may be cleared by the above operation, if called in
+		 * timer irq, no such a need
+		 */
+		if (!z_arc_v2_irq_unit_is_in_isr() ||
```

Comment:
I think it would be better to not call elapsed() in the timer ISR, but duplicate the relevant part of this code there. Makes things easier to understand and less complex (avoids this check that I can't oversee is always correct).

Reply:
good idea

Reply:
elapsed() may be called multiple in timer_int_handler, one is at the beginning,  others are called in z_clock_announce -> z_clock_set_timeout.

So better to handle it in elapsed()

Reply:
Understood. Still, I'm not sure this solution is robust, e.g. if another interrupt handler calls elapsed() (reading the timer to get a timestamp for the input data).
And, I'd like to know why the pulse interrupts don't work on IoTDK.